### ISSUE TITLE
Clearly accessible symbol causes inaccessibility error on declaration emit

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#21367]
```ts
export const x = Symbol();
export const y = Symbol();
declare function rand(): boolean;
export function f() {
    return rand() ? x : y;
}
```

Compile with declaration emit on.

**Expected:** No error, the following declaration file:

```ts
export declare const x: unique symbol;
export declare const y: unique symbol;
export declare function f(): typeof x | typeof y;
```

**Actual:**
```
error TS2527: The inferred type of 'f' references an inaccessible 'unique symbol' type. A type annotation is necessary.`
```
